pkg/expr/sql/parser.go

```diff
@@ -1,89 +1,58 @@
 package sql
 
 import (
-	"encoding/json"
 	"fmt"
 	"sort"
-	"strings"
 
+	"github.com/dolthub/vitess/go/vt/sqlparser"
 	"github.com/grafana/grafana/pkg/infra/log"
-	"github.com/jeremywohl/flatten"
-)
-
-const (
-	TABLE_NAME    = "table_name"
-	ERROR         = ".error"
-	ERROR_MESSAGE = ".error_message"
 )
 
 var logger = log.New("sql_expr")
 
 // TablesList returns a list of tables for the sql statement
 func TablesList(rawSQL string) ([]string, error) {
-	duckDB := NewInMemoryDB()
-	rawSQL = strings.Replace(rawSQL, "'", "''", -1)
-	cmd := fmt.Sprintf("SELECT json_serialize_sql('%s')", rawSQL)
-	ret, err := duckDB.RunCommands([]string{cmd})
-	if err != nil {
-		logger.Error("error serializing sql", "error", err.Error(), "sql", rawSQL, "cmd", cmd)
-		return nil, fmt.Errorf("error serializing sql: %s", err.Error())
-	}
-
-	ast := []map[string]any{}
-	err = json.Unmarshal([]byte(ret), &ast)
+	stmt, err := sqlparser.Parse(rawSQL)
 	if err != nil {
-		logger.Error("error converting json sql to ast", "error", err.Error(), "ret", ret)
-		return nil, fmt.Errorf("error converting json to ast: %s", err.Error())
+		logger.Error("error parsing sql: %s", err.Error(), "sql", rawSQL)
+		return nil, fmt.Errorf("error parsing sql: %s", err.Error())
 	}
 
-	return tablesFromAST(ast)
-}
+	tables := make(map[string]struct{})
+
+	walkSubtree := func(node sqlparser.SQLNode) {
+		err = sqlparser.Walk(func(node sqlparser.SQLNode) (kontinue bool, err error) {
```

Comment:
```suggestion
		err = sqlparser.Walk(func(node sqlparser.SQLNode) (bool, error) {
```
Generally best to avoid named returned params in go.